lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -5134,6 +5134,15 @@ Op LLVMToSPIRVBase::transBoolOpCode(SPIRVValue *Opn, Op OC) {
 SPIRVInstruction *
 LLVMToSPIRVBase::transBuiltinToInstWithoutDecoration(Op OC, CallInst *CI,
                                                      SPIRVBasicBlock *BB) {
+  if (BM->isAllowedToUseVersion(VersionNumber::SPIRV_1_4)) {
+    // OpAtomicCompareExchangeWeak is not "weak" at all,
+    // but instead has the same semantics as OpAtomicCompareExchange.
+    // Moreover, OpAtomicCompareExchangeWeak has been deprecated for
+    // SPIR-V version 1.4 or later.
+    if (OC == OpAtomicCompareExchangeWeak) {
+      OC = OpAtomicCompareExchange;
+    }
```

Comment:
nit: LLVM's coding guidelines suggest not to separate one-liners with brackets
```suggestion
    if (OC == OpAtomicCompareExchangeWeak)
      OC = OpAtomicCompareExchange;
```

Reply:
updated